### PR TITLE
Post release updates

### PR DESCRIPTION
**Pull request by bollwyvl**
- Rework readme
- bump all the versions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,52 +1,124 @@
...
+was:
+- easy to start, to that we could get to the Good Stuff
+- cross-platform, so that everyone could follow along
+- self-contained, so that it could be easily removed
+- offline-capable, so that we could handle intermittent network access
+- hackable by the user, so that we could fix Surprise Features
+
+> ### _Field Notes_
+>
+> _At our 2019 workshop, graciously hosted by [Silli][], (whose network was super fast!)
```
Comment (datakurre) [inline README.md]: Siili, not silli

(both are animals in Finnish, but very different :)

Comment (bollwyvl) [reply README.md]: 104b5e0  !

**Comment by bollwyvl** [comment]
K, cleared up that test. Please update with any other stuff you think it needs... or just (squash) merge!

**Comment by datakurre** [comment]
Thanks. Something to add later is upgrade instructions. I don’t even know, what’s the recommended version. Running a new installer requires uninstalling  the previous version at first.

**Comment by bollwyvl** [comment]
Thanks for merging.

Yeah, theoretically, upgrading is just a matter of picking the old install
location. There's no particular reason it shouldn't work. Practically,
however, deleting and re-adding it seems to be more predictable.

One thing that won't be covered by the install instructions are the
.jupyter directories (the windows one is particularly hard to find) that
would keep certain files around, e.g. jupyterlab workspaces.

To somewhat understand it, now that we have a release we kinda like, we
could have a (separate) test that runs the released installer, runs the
(old) tests (as checked in at that tag), then installs the just-built
installer on top of it, and run the tests there? Seems like a lotta crazy,
but probably worth it, even if it finds one bug.

**Comment by datakurre** [comment]
Currently the installer doesn’t allow replace the old installation, but following the uninstallation instructions and installing then works fine.

I don’t think anything that would require testing for .jupyter directory yet.

How much work would it require to allow upgrades with conda? Custom channel with everything released somewhere around conda forge?

**Comment by bollwyvl** [comment]
Yeah have been thinking about the "proper" in-place upgrade path.

The ideal would indeed be to get everything into conda-forge. I think
everything in the last release is probably now ready to be turned into a
feedstock: it's basically data-driven at this point.

Robotlab, the package, is a little tricky. We're using it to do most of the
pinning now, so it (and its dependencies) can't really be upgraded in
place, and hard pins things like jupyterlab.

The anaconda guys have an approach where you do a package of hard pins and
one of soft pins (just bottom versions, usually). If you don't uninstall
the hard pins, you can install anything your want, as long as it doesn't
change what they've pinned.

You can uninstall the hard pins, and leave the soft pins, allowing you to
upgrade stuff, but leaving the whole system (probably) working.

Or your can uninstall both, and do whatever you want: at that point, I
would just recommend using miniconda and environment files.

**Comment by datakurre** [comment]
Why wouldn't it work to have hard pins in robotlab package and then release a new version of the package with new pins to initiate the upgrade?

**Comment by bollwyvl** [comment]
Yeah, that would work.

For a pre-existing install to find it, we'd actually have to publish
robotlab (the package) and all of its dependencies someplace, and it's kind
of messy (for the reasons described above). While, as I mentioned, our
upstreams area now mostly solid, it would probably not be appropriate to
publish robotlab itself on, say, conda-forge, so we'd have to make our own
channel someplace (anaconda.org, azure, or GitHub, somehow) and bake that
into the installer.

Then there's the activity of finding it... We could put a launcher button
in lab, but that also gets kind of messy, as python processes don't like
having their binderies changed out from underneath them, typically.

Anyhow, certainly food for thought!

**Comment by datakurre** [comment]
Thanks. Now this makes sense. For now “uninstall before reinstall” is enough, but it is good to have a roadmap :) Maybe a use-case for GitHub package registry ...

**Pull request merged**